### ISSUE TITLE
Accuracy difference between models trained on TPU vs GPU

### ISSUE THREAD

**Issue by grassknoted** [pytorch/xla#5755]
## ❓ Questions and Help

I'm training a `vit_base_patch16` on a TPU VM (v3-256), and I notice a very substantial difference in accuracy between the same checkpoints from on GPU and TPU.

**Experiment:**
To test this, I load the Official ImageNet-22k weights from [here](https://console.cloud.google.com/storage/browser/_details/vit_models/imagenet21k/ViT-B_16.npz;tab=live_object), and test on a Linear Probing task on GPU.

On a `v3-256` TPU, I run the following code to create a `vit_base_patch16` model and load the same ImageNet-22k weights, and save a checkpoint without any training at all:
```
import models_vit
import torch_xla.core.xla_model

# Loading the VIT Model
model = models_vit.__dict__["vit_base_patch16"](num_classes = n)

# Loading ImageNet-22K weights from the checkpoint
checkpoint = torch.load(<checkpoint_path>, map_location='cpu')
model.load_state_dict(checkpoint['model'], strict=False)

# Saving the checkpoint without any training
to_save = {
                'model': model.state_dict(),
                'optimizer': optimizer.state_dict(),
                'epoch': epoch,
                'args': args,
            }
torch_xla.core.xla_model.save(to_save, <TPU_checkpoint_path>, global_master=True)
```

Now, I run the Linear Probing task on a GPU with the weights saved in the `<TPU_checkpoint_path>` from the code above, and as a control I load the model directly from [here](https://console.cloud.google.com/storage/browser/_details/vit_models/imagenet21k/ViT-B_16.npz;tab=live_object) on the GPU using the same commands. The results are as follows:

| Checkpoint Source                                                     | Top-1 Acc. | Top-5 Acc. |
| -------------------------------------------------- |----------- | ----------- |
| ImageNet-22k weights directly on GPU |   95.228    |     99.247    |
| Same weights loaded and saved on TPU  |  83.021    |    95.336     |

Ideally, these are the same weights, and the Top-1 accuracy and Top-5 accuracy should be comparable.

I've tried looking for resources online to explain this difference, but I haven't found anything yet.

Additionally, I've tried setting `XLA_USE_BF16`, to both `0` and `1`, but that causes a very slight change in the accuracy for the same seeds. 

**GPU Details:**
```
GPU Device: NVIDIA A40
```
```
$ nvcc -V
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2022 NVIDIA Corporation
Built on Wed_Jun__8_16:49:14_PDT_2022
Cuda compilation tools, release 11.7, V11.7.99
Build cuda_11.7.r11.7/compiler.31442593_0
```

Any help on this would be greatly appreciated!

**Comment by JackCaoG** [comment]
on TPU the matmul is by default execute on bf16 inputs, meaning even if the tensor is f32, during execution it will be downcast to bf16. If you run your GPU mode(I assume on eager pytorch) on bf16 weights would it give similar results?
